src/test/model/node.test.ts

```diff
@@ -0,0 +1,121 @@
+import assert from 'assert';
+
+import { GraphLoader } from '../../main/runtime/loader.js';
+import { runtime } from '../runtime.js';
+
+describe('Node', () => {
+
+    describe('canLinkTo', () => {
+
+        it('specifies whether a property is linkable or not', async () => {
+            const loader = new GraphLoader();
+            const graph = await loader.loadGraph({
+                nodes: [
+                    {
+                        id: 'res',
+                        ref: 'string',
+                        props: [
+                            { key: 'value', value: '42' },
+                        ]
+                    },
+                    {
+                        id: 'p',
+                        ref: 'string',
+                        props: [
+                            { key: 'value', value: '42', linkId: 'res' },
+                        ]
+                    }
+                ],
+                refs: {
+                    string: runtime.defs['string'],
+                }
+            });
+            const node1 = graph.getNodeById('res');
+            const node2 = graph.getNodeById('p');
+            const res1 = node2 ? node1?.canLinkTo(node2) : null;
```

Comment:
In tests like these I'd actually prefer sticking an `!` few lines above so that we're not bothered by null checks (i.e. so that both `node1` and `node2` are `Node` and not `Node | null`.
You can also inline this variable into `assert` because it's used only once — but that's entirely up to you.